### DISCUSSION TITLE
"Cheapest hour calendar" or "AIO energy price hours calendar"

### DISCUSSION THREAD

**Discussion by typxxi** [kotope/aio_energy_management#30] [Q&A]
I was looking at my entity page for "aio" and saw this **"calendar.kalender_gunstiger_strompreise_aio"**
aka cheapest hours calendar.

It made me wonder if that would really be a cheapest hours calendar or if there would be another one maybe missing, that for the most expensive hour.

But both appeared in this calendar and therefore I would suggest to call that the "AiO energy prices calendar"

It would make clear what it is about and if it is filtering or not cause. But I guess the main filter in such case would be the yaml file and then "deactivating" the entity?

But how would I do that the elegant way ?
number_of_hours = 0

I guess. 

```
    - nordpool_entity: sensor.nordpool_kwh_ger_eur_4_10_0
      unique_id: aio_tibbers_expensive_hours
      name: TIBBERS TEUERSTE STUNDEN [AIO]
      first_hour: 14
      last_hour: 13
      starting_today: true
      number_of_hours: 0
      sequential: False
      failsafe_starting_hour: 22
      inversed: true
```

I know that it might not work for you to change elements this way once the integration has been deployed and that you can not rename elements this way, but still a point to know that an element deserves the best name to avoid question marks like "if it is really the calendar for the cheapest hours where is the calendar for the most expensive hours".

I could understand such title based on an integration called "cheapest hours" but that has changed to AIO but the calendar name has not changed with that. 

Sun is now here, all sensors are there, so I am waiting for a cheapest hour to arrive and to turn on the sensor.

**Comment by kotope** [comment]
The calendar entity name is generated by Home Assistant itself from the name you provide it.
There's only one calendar in the whole AIO Energy Management and currently no possibility to add more than one.

If you rename the calendar you could get another one and the previous is coming obsolete that you need to delete manually from Home Assistant itself.
There could be something in Home Assistant APIs to make it better, but I haven't dug into that deep yet. Unfortunately I only have that much free time to develop this so I have to do one thing at a time :-)
